### PR TITLE
executor: migrate test-infra to testify for executor/brie_test.go

### LINKED ISSUE #28568: migrate test-infra to testify for `brie_test.go`

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #28568

Problem Summary:

### What is changed and how it works?

migrate test-infra to testify for executor/brie_test.go

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by hawkingrei** [comment]
PTAL

/cc @tisonkun @AilinKid @TennyZhuang

**Comment by TennyZhuang** [comment]
LGTM, but I’m not reviewer of TiDB.

File: executor/brie_test.go [hunk 1]
```diff
@@ -18,32 +18,29 @@ import (
...
+func TestGlueGetVersion(t *testing.T) {
+	t.Parallel()
 	g := tidbGlueSession{}
 	version := g.GetVersion()
-	c.Assert(version, Matches, `(.|\n)*Release Version(.|\n)*`)
-	c.Assert(version, Matches, `(.|\n)*Git Commit Hash(.|\n)*`)
-	c.Assert(version, Matches, `(.|\n)*GoVersion(.|\n)*`)
+	require.Regexp(t, `(.|\n)*Release Version(.|\n)*`, version)
+	require.Regexp(t, `(.|\n)*Git Commit Hash(.|\n)*`, version)
+	require.Regexp(t, `(.|\n)*GoVersion(.|\n)*`, version)
```
Comment (kennytm) [inline executor/brie_test.go:43]: use `require.Contains` for these

File: executor/brie_test.go [hunk 2]
```diff
@@ -63,13 +60,14 @@ func brieTaskInfoToResult(info *brieTaskInfo) string {
 	return strings.Join(arr, ", ") + "\n"
 }
 
-func fetchShowBRIEResult(c *C, e *ShowExec, brieColTypes []*types.FieldType) string {
+func fetchShowBRIEResult(t *testing.T, e *ShowExec, brieColTypes []*types.FieldType) string {
 	e.result = newFirstChunk(e)
-	c.Assert(e.fetchShowBRIE(ast.BRIEKindBackup), IsNil)
+	require.Nil(t, e.fetchShowBRIE(ast.BRIEKindBackup))
```
Comment (kennytm) [inline executor/brie_test.go:65]:
```suggestion
	require.NoError(t, e.fetchShowBRIE(ast.BRIEKindBackup))
```

File: executor/brie_test.go [hunk 3]
```diff
@@ -95,20 +93,20 @@ func (s *testBRIESuite) TestFetchShowBRIE(c *C) {
...
+	require.NoError(t, err)
 	schema := plan.Schema()
 
 	// Compose executor.
 	e := &ShowExec{
 		baseExecutor: newBaseExecutor(sctx, schema, 0),
 		Tp:           ast.ShowBackups,
 	}
-	c.Assert(e.Open(ctx), IsNil)
+	require.Nil(t, e.Open(ctx))
```
Comment (kennytm) [inline executor/brie_test.go:106]:
```suggestion
	require.NoError(t, e.Open(ctx))
```

**Review by kennytm — COMMENTED** [review]
rest LGTM

File: executor/brie_test.go (hunk 1 above)
Comment (hawkingrei) [reply executor/brie_test.go]: Have fixed.

**Review by kennytm — APPROVED** [review]

**Comment by hawkingrei** [comment]
PTAL @tisonkun @AilinKid @kennytm

**Review by time-and-fate — APPROVED** [review]

**Pull request merged**